### PR TITLE
In-Out-FIle wrappers added

### PR DESCRIPTION
**Pull request by Chupakabra0**

### CODE REVIEW

File: utils/include/CDummyPointer.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#pragma once
+#include <atomic>
+
+template<typename Type>
+class CDummyPointer {
```
Comment (AntonVoytov) [inline utils/include/CDummyPointer.h:5]: What is the point of this class?

File: utils/include/CDummyPointer.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+	pointer operator->() const {
+		return pointer_;
+	}
+
+private:
+	pointer pointer_;
+	std::atomic_uint* counter_;
+
+	bool TryToDelete() {
```
Comment (AntonVoytov) [inline utils/include/CDummyPointer.h:66]: First methods then fileds

**Review by AntonVoytov — COMMENTED** [review]
In general overwhelming functionality, think about how it can be reduced.

**Comment by Chupakabra0** [comment]
1. Removed realization of "smart pointer", added C-pointert `FILE`;
2. Removed a lot of interim interfaces.

**Review by AntonVoytov — APPROVED** [review]

**Pull request merged**